Implement no modal workflow for swap/limit/pool confirmation

Jadessu opened the pull request:
### Describe your changes 
_Describe the purpose + content of these changes_
Implement no modal workflow for swap confirmation.
Test only functionality. For now, please disregard any UI or verbiage inconsistencies
<img width="562" alt="image" src="https://github.com/CrocSwap/ambient-ts-app/assets/83131501/69dc8052-5282-47d3-80f3-9e8184f18b0c">
<img width="580" alt="image" src="https://github.com/CrocSwap/ambient-ts-app/assets/83131501/35374497-6bb0-4646-a32c-177c86c6e850">
<img width="650" alt="image" src="https://github.com/CrocSwap/ambient-ts-app/assets/83131501/104d71a7-003e-4d85-828e-89b85b76c4d5">



### Checklist before requesting a review
- [ ] Is this PR ready for merge? (Please convert to a draft PR otherwise)
- [x] I have performed a self-review of my code.
- [x] Did I request feedback from a team member prior to merge? 
- [x] Does my code following the style guide at `docs/CODING-STYLE.md`?

### Instructions for Reviewer
**Functionalities or workflows that should specifically be tested.**
Test only functionality. For now, please disregard any UI or verbiage inconsistencies

Jadessu committed:
refactor trade module header

Jadessu committed:
to resume

Jadessu committed:
Merge branch 'develop' of https://github.com/CrocSwap/ambient-ts-app into implement-stepper-confirmation-swap

Jadessu committed:
add tokens display

Jadessu committed:
to resume

Jadessu committed:
clean up unused variables

Jadessu committed:
Merge branch 'develop' of https://github.com/CrocSwap/ambient-ts-app into implement-stepper-confirmation-swap

Jadessu committed:
Merge branch 'develop' of https://github.com/CrocSwap/ambient-ts-app into implement-stepper-confirmation-swap

Jadessu committed:
implement swap confirmation logic

Jadessu committed:
add conf steps for limit order

Jadessu committed:
fix lint errors

Jadessu committed:
to resume

Jadessu committed:
Merge branch 'develop' of https://github.com/CrocSwap/ambient-ts-app into implement-stepper-confirmation-swap

Jadessu committed:
attempt to fix bug

surudhb committed:
Merge branch 'develop' into implement-stepper-confirmation-swap

surudhb committed:
merge develop

surudhb committed:
reset showstepperconfirmation on confirmation

Jadessu committed:
change steps text

benwolski committed:
Merge branch 'develop' into implement-stepper-confirmation-swap

Jadessu committed:
add reset function to back button

Jadessu committed:
Merge branch 'implement-stepper-confirmation-swap' of https://github.com/CrocSwap/ambient-ts-app into implement-stepper-confirmation-swap

Jadessu committed:
add reset function to transaction settings

benwolski committed:
update verbiage of trade confirmations

Jadessu committed:
add pool price to trade confirmation

Jadessu committed:
add pool token info to confirmation

Jadessu committed:
add dropdown arrow to control wallet buttons in transactions

Jadessu committed:
remove dropdown and console logs

miyuki-eto commented:
a few comments on this -

1. the submit/confirm/send another buttons have inconsistent widths between different screens
2. the first confirmation screen is missing the arrow between assets/qtys
3. the spacing on the final confirmed screen needs adjusting so its equal (assets with arrows, swap success, view on explorer, import to wallet)
4. ideally the submit/confirm/send another buttons would all be in the same location. this way the user can click through without moving their mouse at all

![image](https://github.com/CrocSwap/ambient-ts-app/assets/82160202/c85feb34-807f-4f5d-9156-7540719db5f1)

Jadessu committed:
fix sizing and rendering issue for swap page

Jadessu committed:
add min size for mobile devices

Jadessu commented:
> a few comments on this -
> 
> 1. the submit/confirm/send another buttons have inconsistent widths between different screens
> 2. the first confirmation screen is missing the arrow between assets/qtys
> 3. the spacing on the final confirmed screen needs adjusting so its equal (assets with arrows, swap success, view on explorer, import to wallet)
> 4. ideally the submit/confirm/send another buttons would all be in the same location. this way the user can click through without moving their mouse at all
> 
> ![image](https://private-user-images.githubusercontent.com/82160202/288637835-c85feb34-807f-4f5d-9156-7540719db5f1.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTEiLCJleHAiOjE3MDE5ODU3OTIsIm5iZiI6MTcwMTk4NTQ5MiwicGF0aCI6Ii84MjE2MDIwMi8yODg2Mzc4MzUtYzg1ZmViMzQtODA3Zi00ZjVkLTkxNTYtNzU0MDcxOWRiNWYxLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFJV05KWUFYNENTVkVINTNBJTJGMjAyMzEyMDclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjMxMjA3VDIxNDQ1MlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWJhM2ZkNmUxYTU1MmNiMmFlNzNjOWYwMDYwNWNiN2UyMjdiNzY5OGZjNWYzYWM5YTdkZmIzYzlkMmE2YjZjNzgmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.8CpH9P4dNKi4pUA4Fz8h70a5RQ9ImOvdNpubJEsDf4A)



@miyuki-eto @benwolski  I've implemented the requested changes and wanted to provide some insights:

Inconsistent Button Widths:
In the Figma mockup, the submit/confirm/send another buttons had varying widths(Not as exaggerated as what I had implemented). To maintain consistency now, I've set a padding of 32px on each side for all buttons. If you prefer a different value, we can adjust accordingly.
| Step 1                                              | Step 2                                               | Step 3                                               |
| ------------------------------------------------------ | ------------------------------------------------------ | ------------------------------------------------------ |
| ![Image 1](https://github.com/CrocSwap/ambient-ts-app/assets/83131501/a38fafb8-f6f3-4d0b-b906-625c698c5e11) | ![Image 2](https://github.com/CrocSwap/ambient-ts-app/assets/83131501/72479861-acc8-4b5f-9ddc-fe7e2c958287) | ![Image 3](https://github.com/CrocSwap/ambient-ts-app/assets/83131501/2353a4f8-0387-4896-b2ef-995aed80862c) |


Button Placement:
 I've refactored and kept the submit/confirm/send another buttons in the same location across all steps. In order to achieve this, the container height is set to 450px to accommodate all content. Note that opening the extraInfo dropdown on the first step may change the height of the first step to acomodate the new added data.

Regarding this, we have two options:

Option 1: Increase the container height to 600px to accommodate everything from the start. This ensures the button is always in the same place but results in additional empty space.
| Before                                      | After                                       |
| -------------------------------------------- | -------------------------------------------- |
| ![Before Image 1](https://github.com/CrocSwap/ambient-ts-app/assets/83131501/bbab4ac5-ba9b-4a02-95cf-2c9a104f1c66) | ![After Image 1](https://github.com/CrocSwap/ambient-ts-app/assets/83131501/611adcde-d4d9-49b7-ae20-db6e3fd0301e) |
| ![Before Image 2](https://github.com/CrocSwap/ambient-ts-app/assets/83131501/19c5ad85-b4ee-4f57-8195-1d73aa230993) | ![After Image 2](https://github.com/CrocSwap/ambient-ts-app/assets/83131501/34f61f68-d6ed-4111-ae85-4b558689019c) |



Current Implementation: Keep the container height at 450px, adjusting dynamically based on user actions. This minimizes empty space but may result in a slight shift in button position when the extraInfo dropdown is open.

I personally recommend the current implementation for its minimalistic approach and maintaining the same button position until user interaction necessitates a change in container height.

Let me know your thoughts.

miyuki-eto commented:
thanks junior. if there are ever inconsistencies in the design files feel free to let me know and i can update them, sometimes they can be easy to miss. and yeah i agree the second option is better. use the consistent button placement based on the 450px height and if the user expands it it can be out of line

Jadessu committed:
make swap conf more compact

Jadessu committed:
add size to trade modules

Jadessu committed:
add size to confirm messages

Jadessu committed:
final push

benwolski committed:
Merge branch 'develop' of https://github.com/CrocSwap/ambient-ts-app into implement-stepper-confirmation-swap

Jadessu committed:
M"

Jadessu committed:
fix pool price display in confirmation

benwolski committed:
update skip confirmation verbiage

benwolski committed:
update limit removal explanation verbiage

benwolski committed:
Merge branch 'develop' into implement-stepper-confirmation-swap

benwolski committed:
Merge branch 'develop' of https://github.com/CrocSwap/ambient-ts-app into implement-stepper-confirmation-swap

benwolski committed:
set cursor for header to default during confirmation

benwolski commented:
marking draft until a determination is made as to whether limit/range updates should be prevented during confirmations

Jadessu committed:
add active confirmation context

Jadessu committed:
remove chart disabling and fix activation parameters

Jadessu committed:
change limit order height

Jadessu committed:
remove unused variables

tncoskun committed:
disabled drag and click for limit and range lines/labels according to isConfirmationActive

benwolski committed:
Merge pull request #3299 from tncoskun/disabling-dragging-limit-and-range-lines-until-the-trades-completed

Chart : disabled drag and click for limit and range lines/labels according to isConfirmationActive

Jadessu committed:
deactivate confirmation on link change

benwolski committed:
Merge branch 'develop' into implement-stepper-confirmation-swap

benwolski committed:
Merge branch 'develop' into implement-stepper-confirmation-swap

Jadessu committed:
fix resize issue when extra info dropdown is active

Jadessu committed:
Merge branch 'implement-stepper-confirmation-swap' of https://github.com/CrocSwap/ambient-ts-app into implement-stepper-confirmation-swap

benwolski commented:
currently working on the Reposition stepper confirmation

Jadessu committed:
create reposition stepper layout

Jadessu committed:
create no modal transaction settings

Jadessu committed:
refactor transactions settings to not use modal in trade modules

Jadessu committed:
remove modal from share component on trade

Jadessu committed:
add dynamic token data to reposition

tncoskun committed:
limit and range lines tooltip added for confirmation

Jadessu committed:
fix merge conflicts

benwolski committed:
Merge branch 'develop' into implement-stepper-confirmation-swap

Commit:
Merge branch 'develop' of github.com:CrocSwap/ambient-ts-app into implement-stepper-confirmation-swap

tncoskun committed:
chartLinesTooltip css changed

benwolski committed:
Merge pull request #3306 from tncoskun/chart-lines-tooltip-implement-stepper-confirmation-swap

Tooltip Added for Limit and Range Lines During Confirmation

Commit:
update disabled feature tooltip verbiage

Jadessu committed:
fix lint error

Jadessu committed:
Merge branch 'implement-stepper-confirmation-swap' of https://github.com/CrocSwap/ambient-ts-app into implement-stepper-confirmation-swap

benwolski commented:
@Jadessu, the confirm button is disappearing (need to scroll to see) when I open the extra info section of the pool moduleon my laptop, even when there appears to be sufficient space to display it in advanced mode:

![image](https://github.com/CrocSwap/ambient-ts-app/assets/570819/a9a8ee68-b5b5-49ed-85e9-dda2754a0b25)

Jadessu committed:
add handle go back function to go to main

Jadessu committed:
remove settings on reposition

Jadessu committed:
fix quantity container overflow issue

Jadessu committed:
replace arrow with dash

benwolski committed:
Merge branch 'develop' into implement-stepper-confirmation-swap

Jadessu committed:
update module sizing

benwolski committed:
Merge branch 'develop' into implement-stepper-confirmation-swap

benwolski committed:
Merge branch 'develop' into implement-stepper-confirmation-swap

benwolski committed:
Merge branch 'develop' into implement-stepper-confirmation-swap

benwolski committed:
Merge branch 'develop' into implement-stepper-confirmation-swap

Jadessu committed:
remove dash on pool-reposition confirmation

benwolski committed:
Merge branch 'develop' of https://github.com/CrocSwap/ambient-ts-app into implement-stepper-confirmation-swap

Jadessu committed:
add cursor to reposition

benwolski commented:
@Jadessu, we'll need to add the price impact warning on the new swap confirmation as well

Jadessu committed:
add price impact warning to swap confirmation

benwolski committed:
Merge branch 'develop' of https://github.com/CrocSwap/ambient-ts-app into implement-stepper-confirmation-swap

benwolski committed:
Merge branch 'develop' into implement-stepper-confirmation-swap

Jadessu committed:
refactor chip to disabled on hover

Jadessu committed:
Merge branch 'implement-stepper-confirmation-swap' of https://github.com/CrocSwap/ambient-ts-app into implement-stepper-confirmation-swap

tncoskun committed:
fixed : disable drag limit when start confirmation

benwolski committed:
Merge pull request #3336 from tncoskun/check-conflict-implement-stepper-confirmation-swap

fixed : disable drag limit when start confirmation

benwolski committed:
Merge branch 'develop' into implement-stepper-confirmation-swap

benwolski committed:
prevent  token quantity updates after swap

benwolski committed:
Merge branch 'develop' of https://github.com/CrocSwap/ambient-ts-app into implement-stepper-confirmation-swap

benwolski committed:
Merge branch 'implement-stepper-confirmation-swap' into fix-no-modal-confirmation-displays

benwolski committed:
Merge pull request #3338 from CrocSwap/fix-no-modal-confirmation-displays

prevent  token quantity updates after swap (no-modal)

benwolski committed:
Merge branch 'develop' into implement-stepper-confirmation-swap

benwolski committed:
Merge branch 'develop' of https://github.com/CrocSwap/ambient-ts-app into implement-stepper-confirmation-swap

benwolski committed:
Merge branch 'develop' of https://github.com/CrocSwap/ambient-ts-app into implement-stepper-confirmation-swap

benwolski committed:
Merge branch 'develop' into implement-stepper-confirmation-swap

benwolski committed:
Merge branch 'develop' into implement-stepper-confirmation-swap

benwolski committed:
Merge branch 'develop' of https://github.com/CrocSwap/ambient-ts-app into implement-stepper-confirmation-swap

miyuki-eto reviewed (changes requested):
a few notes - 

see figma here for annotations - https://www.figma.com/file/0ItLk4nJmtj741R8sn3HYp/ambient?type=design&node-id=8865-56580&mode=design&t=rLd73m0IxBdVMaDP-4

1. on swap page the background image doesnt take up the whole space. it stops at the bottom of the swap box
2. is the token selector set up for longer ticker names? it looks like if the ticker is 7+ characters long there is spacing issues or the arrow disappears
3. can we use chevron arrows instead of the back arrows currently implemented?
4. increase the space between the top of the swap module and the header text
5. swap confirmation header isnt centered
6. extra info box on swap confirmation page doesnt have the right spacing and margins
7. can the front run warning on the settings page have centered text
8. swap settings submit button location doesnt line up with buttons on other pages
9. share page has an extra purple hover when hovering over the twitter etc logos
10. increase the spacing between the confirmation steps and loading spinners
11. remove the dark 2 colour inside the loading spinner
12. can we keep the distance between the header and the spinners/step consistent across each stage? currently the last stage moves the steps up
13. the select token modal should be incorporated into the module - refer figma
14. does there have to be so much space below the extra info section and the confirm buttons? it doesnt look like that space is necessary for all the other pages
15. trade module - increase the space between header and content
16. do the token logos have to be so centered on the confirm pages?
17. pool confirm - can the pool details at the top follow what currently shown on he chart? now if i swap the tokens while doing a swap (eg swapping usdc to eth) that follows through to the pool confirm page and doesnt match whats shown in the chart area

benwolski committed:
Merge branch 'develop' of https://github.com/CrocSwap/ambient-ts-app into implement-stepper-confirmation-swap

benwolski committed:
resolve no-modal confirmation branch merge conflicts

tncoskun committed:
fixed: clicking on canvas for range/limit

benwolski committed:
Merge pull request #3387 from tncoskun/fix-lines-drag-resolve-no-modal-merge-conflicts

fixed: clicking on canvas for range/limit

benwolski committed:
Merge pull request #3386 from CrocSwap/resolve-no-modal-merge-conflicts

resolve no-modal confirmation branch merge conflicts

Commit:
Merge branch 'develop' of github.com:CrocSwap/ambient-ts-app into implement-stepper-confirmation-swap
